.editorconfig

```diff
@@ -17,3 +17,20 @@ dotnet_sort_system_directives_first = true
 
 dotnet_diagnostic.IDE0073.severity = warning  # Require file header
 file_header_template = Copyright (c) Duende Software. All rights reserved.\nSee LICENSE in the project root for license information.
+
+dotnet_diagnostic.IDE0161.severity = error  # Use file-scoped namespace
+csharp_style_namespace_declarations = file_scoped:error
+
+dotnet_diagnostic.IDE0003.severity = error # Enforce that 'this.' should not be used when it's redundant
+
+dotnet_diagnostic.IDE0049.severity = error # Enforce using built-in type keywords (e.g., 'string' instead of 'String')
+
+csharp_new_line_before_open_brace = all # BSD/Allman braces: Place braces on a new line for all constructs
+dotnet_diagnostic.IDE0011.severity = error # Require braces for all if statements
+dotnet_diagnostic.IDE0055.severity = error # Keep empty braces { } on the same line
+csharp_style_allow_blank_line_after_open_brace = false
+csharp_style_allow_blank_line_between_consecutive_braces = false
+
+csharp_style_var_elsewhere = true:error # Enforce 'var' usage for all local variables
+csharp_style_var_when_type_is_apparent = true:error
+csharp_style_var_for_built_in_types = true:error
```

Comment:
These are options for IDE0007, so they should appear below a setting for that rule.